pkg/service/service.go

```diff
@@ -176,36 +178,83 @@ func (s *Service) GetCurrentTs() time.Time {
 	return time.Now()
 }
 
+func (s *Service) populateV2BeMapFromV1BeMap(ipv4, ipv6 bool) error {
+	removeMap := func(m string) {
+                p := path.Join(bpf.MapPrefixPath(), m)
+                if _, err := os.Stat(p); !os.IsNotExist(err) {
+                        if e := os.RemoveAll(p); e != nil {
+                                log.WithError(err).WithField(logfields.Path, p).Warn("Error deleting map")
+                        } else {
+                                log.WithField(logfields.Path, p).Info("Deleted bpf map")
+                        }
+                }
+        }
+	if ipv4 {
+		copyBackend4Entries := func(key bpf.MapKey, value bpf.MapValue) {
+			backendKey := key.(lbmap.BackendKey)
+			backend4MapKeyV2 := lbmap.NewBackend4KeyV2(backendKey.GetID())
+			lbmap.Backend4MapV2.Update(backend4MapKeyV2, value.DeepCopyMapValue())
+		}
+		err := lbmap.Backend4Map.DumpWithCallback(copyBackend4Entries)
+		if err != nil {
+			return fmt.Errorf("Unable to populate v2 lb4 backends map: %s", err)
+		}
+		lbmap.Backend4Map.Close()
+		removeMap(lbmap.Backend4MapName)
```

Comment:
Maybe add a comment saying that the map will be actually removed once the last program referencing it has been removed.

Reply:
It will be removed from the fs at this point, correct? So it would still be in-memory but if you do another restart, it should be gone completely?

Reply:
The map is referenced by the BPF progs (in our case - bpf_host.o instances). Once the map is unpinned (=removed from the fs) and the last BPF prog referencing is gone, the map will be from memory.

Reply:
I see, thanks! So bpf_host would be reloaed(removed then re-hooked) upon endpoint regeneration? Which endpoint hooks bpf_host though?

Reply:
The host endpoint (`reserved:host`) via `Loader.reloadHostDatapath`.

Reply:
Ah I see, thanks!